core/src/main/java/org/kiwix/kiwixmobile/core/ViewModelFactory.kt

```diff
@@ -0,0 +1,48 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2022 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package org.kiwix.kiwixmobile.core
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import javax.inject.Inject
+import javax.inject.Provider
+
+open class ViewModelFactory @Inject constructor(
+  private val creators: Map<Class<out ViewModel>, @JvmSuppressWildcards Provider<ViewModel>>
```

Comment:
its issue related to Dagger , when we are creating multiple object with different Class type then we need to  provide this otherwise it will give us compilation error https://adavis.info/2017/08/jvmsuppresswildcards-biggest-annoyance-kotlin.html. and its only happening with kotlin while java works fine.

Reply:
Ok. We don't run lint check on Java.